tests/helpers.py

```diff
@@ -201,6 +206,35 @@ def create_pod(
 
     if sa_name and dc_deployment:
         pod_data['spec']['template']['spec']['serviceAccountName'] = sa_name
+
+    # disconnected cluster related configuration updates
+    if config.DEPLOYMENT.get('disconnected'):
+        if 'containers' in pod_data['spec']:
+            container = pod_data['spec']['containers'][0]
+        else:
+            container = pod_data['spec']['template']['spec']['containers'][0]
+        container['image'] = mirror_image(container['image'])
+
+    # configure http[s]_proxy env variable, if required
+    try:
+        if 'http_proxy' in config.ENV_DATA:
+            if 'containers' in pod_data['spec']:
+                container = pod_data['spec']['containers'][0]
+            else:
+                container = pod_data['spec']['template']['spec']['containers'][0]
+            if 'env' not in container:
+                container['env'] = []
+            container['env'].append({
+                'name': 'http_proxy',
+                'value': config.ENV_DATA['http_proxy'],
+            })
+            container['env'].append({
+                'name': 'https_proxy',
+                'value': config.ENV_DATA['http_proxy'],
+            })
+    except KeyError:
+        pass
```

Comment:
hmm, I don't like it :)

Please at least write something in log to warning? It will be hard to debug if some issue with KeyError here.

Reply:
Fixed in 43884ad.